gene2phenotype_project/gene2phenotype_app/views.py

```diff
@@ -898,3 +906,39 @@ def update(self, request, *args, **kwargs):
 
         else:
             return Response({"message": "Failed to update data", "details": serializer.errors})
+
+class PublishRecord(APIView):
+    http_method_names = ['post', 'head']
+    serializer_class = CurationDataSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def post(self, request, stable_id, *args, **kwargs):
+        user = self.request.user
+
+        try:
+            # Get curation record
+            curation_obj = CurationData.objects.get(stable_id__stable_id=stable_id,
+                                                    user__email=user)
+
+            # Check if there is enough data to publish the record
+            locus_obj = self.serializer_class().validate_to_publish(curation_obj)
+
+            # Publish record
+            try:
+                lgd_obj = self.serializer_class(context={'user': user}).publish(curation_obj)
+                # Delete entry from 'curation_data'
+                curation_obj.delete()
```

Comment:
Please do we need to delete the curation_obj ?

Reply:
When the LGD entry is published there is no need to keep the record in the curation table. 
There is a history table for the curation, if we need to check the logs we can use the history table.